jujugui/static/gui/src/app/components/deployment/deployment.js

```diff
@@ -217,15 +185,36 @@ YUI.add('deployment-component', function() {
       @param {String} newComponent The component to switch to.
     */
     _changeActiveComponent: function(newComponent) {
-      var nextProps = this.state;
-      nextProps.activeComponent = newComponent;
-      this.setState(this.generateState(nextProps));
+      this.setState({activeComponent: newComponent});
     },
 
     render: function() {
+      var activeComponent = this.state.activeComponent;
+      var activeChild = this._generateActivePanel();
+      var steps = [{
+        title: 'Deploy',
+        active: activeComponent === 'deployment-summary'
+      }];
       return (
         <div className="deployment-view">
-          {this.state.activeChild.component}
+          <juju.components.DeploymentBar
```

Comment:
I'm not sure it makes sense for this to be here anymore. Is there a reason why it can't be moved into its own separate component, especially now that it's being completely covered? 

The new designs also have the import/export going into the header so the logic around it is going to get simpler.